pkg/minishift/provisioner/minishift_provisioner.go

```diff
@@ -107,6 +142,13 @@ func (provisioner *MinishiftProvisioner) Provision(swarmOptions swarm.Options, a
 	provisioner.EngineOptions = engineOptions
 	swarmOptions.Env = engineOptions.Env
 
+	// Pre-configure the remote machine if generic driver in use
+	if provisioner.Driver.DriverName() == "generic" {
```

Comment:
Is there a better way to collect information about the driver and OS: https://github.com/praveenkumar/minishift/blob/6ecc79dc2bbd529dbc3666e42a36e5327201336c/pkg/minishift/provisioner/minishift_provisioner.go#L204-L208 in a better way?

We are now adding `generic` or other words VM-driver specific code to our provisioner. Is there not a better way?